$OpenBSD: patch-m4_ax_compiler_flags_cflags_m4,v 1.1 2017/01/30 20:55:50 jasper Exp $

Avoid using a compiler option we don't have: -Wno-suggest-attribute
https://savannah.gnu.org/patch/index.php?9243

--- m4/ax_compiler_flags_cflags.m4.orig	Sat Mar 19 17:00:00 2016
+++ m4/ax_compiler_flags_cflags.m4	Tue Jan 24 18:03:41 2017
@@ -50,6 +50,13 @@ AC_DEFUN([AX_COMPILER_FLAGS_CFLAGS],[
         ax_compiler_flags_test=""
     ])
 
+    # Check that -Wno-suggest-attribute=format is supported
+    AX_CHECK_COMPILE_FLAG([-Wno-suggest-attribute=format],[
+        ax_compiler_no_suggest_attribute_flags="-Wno-suggest-attribute=format"
+    ],[
+        ax_compiler_no_suggest_attribute_flags=""
+    ])
+
     # Base flags
     AX_APPEND_COMPILE_FLAGS([ dnl
         -fno-strict-aliasing dnl
@@ -108,7 +115,7 @@ AC_DEFUN([AX_COMPILER_FLAGS_CFLAGS],[
         AX_APPEND_FLAG([-Werror],ax_warn_cflags_variable)
 
         AX_APPEND_COMPILE_FLAGS([ dnl
-            -Wno-suggest-attribute=format dnl
+            [$ax_compiler_no_suggest_attribute_flags] dnl
         ],ax_warn_cflags_variable,[$ax_compiler_flags_test])
     ])
 
